### PR TITLE
implement guitar string frequencies

### PR DESCRIPTION
**Pull request by friedolino78**

### CODE REVIEW

**Comment by fundamental** [comment]
@JohannesLorenz Since you reviewed the previous sympathetic resonance PR, do you have any comments on this one?

**Comment by JohannesLorenz** [comment]
> @JohannesLorenz Since you reviewed the previous sympathetic resonance PR, do you have any comments on this one?

I thought @friedolino78 considered adding more changes :grin: Is it ready to review now?

**Comment by friedolino78** [comment]
> > @JohannesLorenz Since you reviewed the previous sympathetic resonance PR, do you have any comments on this one?
> 
> I thought @friedolino78 considered adding more changes grin Is it ready to review now?

yes, the changes you suggested are in the stringchoir commit.

File: src/Effects/Sympathetic.cpp [hunk 1]
```diff
@@ -213,20 +230,56 @@ void Sympathetic::calcFreqs()
...
+{
+    const float unison_spread_semicent = powf(Punison_frequency_spread / 63.5f, 2.0f) * 25.0f;
+    const float unison_real_spread_up = powf(2.0f, (unison_spread_semicent * 0.5f) / 1200.0f);
+    const float unison_real_spread_down = 1.0f/unison_real_spread_up;
+
+    for(unsigned int i = 0; i < Punison_size*Pstrings; i+=Punison_size)
+    {
+        const float centerFreq = powf(2.0f, (float)i / 36.0f) * baseFreq;
+        const unsigned int stringchoir_size = 
+			i>num_single_strings ? (i>Pstrings-num_triple_strings ? 3 : 2) :1;
```
Comment (JohannesLorenz) [inline src/Effects/Sympathetic.cpp:243]: I thought pianos have 3 strings for the *lowest* keys, not for the highest ones?

Comment (friedolino78) [reply src/Effects/Sympathetic.cpp:243]: take a second look ;-)
the lower strings have more mass. therefor they can store more energy. to compensate that higher notes get more strings.

Comment (JohannesLorenz) [reply src/Effects/Sympathetic.cpp:243]: Oh, you're right, I got this wrong way round :+1:

File: src/Effects/Sympathetic.cpp [hunk 2]
```diff
@@ -213,20 +227,35 @@ void Sympathetic::calcFreqs()
...
+
 unsigned char Sympathetic::getpresetpar(unsigned char npreset, unsigned int npar)
 {
 #define PRESET_SIZE 13
 #define NUM_PRESETS 4
     static const unsigned char presets[NUM_PRESETS][PRESET_SIZE] = {
-        //Vol Pan Q Drive Lev Spr neg lp hp sz  strings note cross
+        //Vol Pan Q Drive Lev Spr neg lp hp sz  strings note
         //Piano 12-String
         {100, 64, 125, 5, 80, 10, 0, 127, 0, 3,   12,  57},
```
[the remark this answers is not in the document]
Comment (friedolino78) [reply src/Effects/Sympathetic.cpp:276]: done

**Comment by JohannesLorenz** [comment]
Note: rPreset macros checked, are OK (6a3cce9)

File: src/Effects/Sympathetic.cpp [hunk 3]
```diff
@@ -288,10 +319,12 @@ void Sympathetic::changepar(int npar, unsigned char value)
             break;
         case 9:
             Punison_size = limit(value, (unsigned char) 1, (unsigned char) 3);
+            if (Punison_size>2) Ppreset=0;
```
[the remark this answers is not in the document]
Comment (friedolino78) [reply src/Effects/Sympathetic.cpp:346]: I am trying to do a refresh in zest. but it doesn't work yet

[the remark this answers is not in the document]
Comment (friedolino78) [reply src/Effects/Sympathetic.cpp:346]: now it works: https://github.com/mruby-zest/mruby-zest-build/pull/85

**Review by JohannesLorenz — APPROVED** [review]

**Pull request merged**